src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_FIGURE_SIZE = 10;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public int figureSize() {
+        return random.nextInt(MAX_FIGURE_SIZE);
+    }
+
+    public Figure whichFigure() {
+        switch (Figures.values()[random.nextInt(Figures.values().length)]) {
+            case CIRCLE:
+                return new Circle(colorSupplier.getRandomColor(), figureSize());
+            case SQUARE:
+                return new Square(colorSupplier.getRandomColor(), figureSize());
+            case RECTANGLE:
+                return new Rectangle(colorSupplier.getRandomColor(), figureSize(), figureSize());
+            case RIGHTTRIANGLE:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        figureSize(), figureSize());
+            default: ISOSCELESTRAPEZOID:
```

Comment:
it would be better if for default case we will return default figure (`getDefaultFigure()` method)